### PR TITLE
Offer to use class and type instead of component

### PR DESCRIPTION
**Pull request by atoulme**
This allows to use the `component` field in changelog, or to use the `class` and `type` fields. This optional use will allow us down the road to formalize changelog per component.

## Related issue ##
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23249

### CODE REVIEW

File: chloggen/internal/chlog/entry.go [hunk 1]
```diff
@@ -82,7 +120,11 @@ func (e Entry) String() string {
 	issueStr := strings.Join(issueStrs, ", ")
 
 	var sb strings.Builder
-	sb.WriteString(fmt.Sprintf("- `%s`: %s (%s)", e.Component, e.Note, issueStr))
+	component := e.Component
```
Comment (pellared) [inline chloggen/internal/chlog/entry.go:123]: I would rather rename it to something like `prefix`

File: chloggen/internal/chlog/entry.go [hunk 2]
```diff
@@ -47,6 +55,15 @@ var changeTypes = []string{
...
 }
 
+var classTypes = []string{
+	Receiver,
+	Exporter,
+	Processor,
+	Connector,
+	Extension,
+	Cmd,
+}
```
Comment (pellared) [inline chloggen/internal/chlog/entry.go:65]: This is tightly coupled to Collector. Should it not be configurable?

**Review by pellared — COMMENTED** [review]
> This optional use will allow us down the road to formalize changelog per component.

It there any issue which describes/discusses the plan/design?

File: chloggen/internal/chlog/entry.go (hunk 2 above)
Comment (atoulme) [reply chloggen/internal/chlog/entry.go:65]: This is indeed tied to the collector, all other uses can keep using the component field just like today.

File: chloggen/internal/chlog/entry.go (hunk 1 above)
Comment (atoulme) [reply chloggen/internal/chlog/entry.go:123]: Not quite sure I follow, this is not used as a prefix, it's still used as the component name if provided.

**Comment by atoulme** [comment]
@pellared I have filed https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23249 to document the plan.

**Comment by atoulme** [comment]
Closing ; will open a new PR with the new approach based on package.

**Pull request closed without merging**